ykrt/src/compile/jitc_yk/jit_ir/mod.rs

```diff
@@ -923,6 +923,32 @@ impl Const {
             Const::Ptr(_) => m.ptr_ty_idx,
         }
     }
+
+    /// If this constant is an integer that can be represented in 64 bits, return it as an `i64`.
+    pub(crate) fn int_to_i64(&self) -> Option<i64> {
+        match self {
+            Const::I8(x) => Some(i64::from(*x)),
+            Const::I16(x) => Some(i64::from(*x)),
+            Const::I32(x) => Some(i64::from(*x)),
+            Const::I64(x) => Some(i64::from(*x)),
+            Const::Ptr(_) => None,
+        }
+    }
+
+    /// Create an integer of the same underlying type and with the value `x`.
+    ///
+    /// # Panics
+    ///
+    /// If `x` doesn't fit into the underlying integer type.
+    pub(crate) fn i64_to_int(&self, x: i64) -> Const {
+        match self {
+            Const::I8(_) => Const::I8(i8::try_from(x).unwrap()),
+            Const::I16(_) => Const::I16(i16::try_from(x).unwrap()),
+            Const::I32(_) => Const::I32(i32::try_from(x).unwrap()),
+            Const::I64(_) => Const::I64(i64::try_from(x).unwrap()),
```

Comment:
I think `try_from` can be removed for the `I64` case? Doesn't matter much.

Reply:
It'll be optimised away in release mode. But this method is going to have to change soon, anyway, I think...